scripts/files/fs_s3.py

```diff
@@ -67,6 +67,96 @@ def read(path: str, needs_credentials: bool = False) -> bytes:
     return file
 
 
+def rename(path: str, new_path: str, needs_credentials: bool = False) -> None:
+    """'Rename' an s3 object: copy an existing object to another with the new key name and delete the old object.
+        The objects have to be in the same bucket.
+
+    Args:
+        path (str): the path of the object to rename.
+        new_path (str): the path with the new name.
+        needs_credentials (bool, optional): Tells if credentials are needed. Defaults to False.
+
+    Raises:
+        Exception: an exception if the path are not in the same bucket
+        ce: a client exception
+        nsb: a no such bucket exception
+        nsk: a no such key exception
+    """
+    start_time = time_in_ms()
+    src_s3_path = parse_path(path)
+    dst_s3_path = parse_path(new_path)
+
+    if src_s3_path.bucket != dst_s3_path.bucket:
+        get_log().error(
+            "rename_s3_different_buckets",
+            path=path,
+            destination=new_path,
+            error="The source and destination path are not in the same bucket",
+        )
+        raise Exception(f"Files {path} and {new_path} are not on the same S3 bucket.")
+
+    s3 = boto3.resource("s3")
+    try:
+        if needs_credentials:
+            s3 = get_session(path).client("s3")
```

Comment:
this is only going to work if the source and targets are the same bucket?

Reply:
Yes